core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -153,23 +149,34 @@ static void checkManifest(Manifest manifest, Path jar) {
         }
     }
 
+    public static void checkVersionFormat(String targetVersion) {
+        if (!targetVersion.matches("^0*[0-9]+(\\.[0-9]+)*$")) {
+            throw new IllegalStateException(
+                    String.format(
+                            Locale.ROOT,
+                            "version string must be a sequence of nonnegative decimal integers separated by \".\"'s and may have leading zeros but was %s",
+                            targetVersion
+                    )
+            );
+        }
+    }
+
     /**
      * Checks that the java specification version {@code targetVersion}
      * required by {@code resource} is compatible with the current installation.
      */
     public static void checkJavaVersion(String resource, String targetVersion) {
```

Comment:
Can this be renamed to checkVersionCompatible or something? otherwise I think its confusing. Alternatively we can combine the two methods.